app/src/app/components/contents/downloads/ImageDisplay.tsx

```diff
@@ -0,0 +1,17 @@
+import { FileMetadata } from "../../../../types";
+import { useFileObjectUrl } from "./hooks/useFileObjectUrl";
+import { ErrorComponent } from "../common/ErrorComponent";
+
+const defaultErrorMessage = "Error loading image file";
+
+interface ImageDisplayProps {
+  file: FileMetadata;
+}
+
+export const ImageDisplay = ({ file }: ImageDisplayProps) => {
+  const { fileObjectUrl, error } = useFileObjectUrl(file);
+
+  if (error) return <ErrorComponent message={defaultErrorMessage} error={error} />;
+
+  return (fileObjectUrl && <img src={fileObjectUrl} alt={file.path} />) || null;
```

Comment:
Can we not assume that fileObjectUrl is truthy here, since we've already checked for error and returned if there was one..?

Reply:
It initializes as undefined, but it makes no difference to the UX to omit this check.